pkgs/tools/security/cryptomator/default.nix

```diff
@@ -0,0 +1,113 @@
+    mimeType="application/x-vnd.cryptomator-vault-metadata";
+    categories = "Utility;Security;FileTools;";
+    terminal = "false";
+    extraEntries = ''
+      StartupWMClass=org-cryptomator-launcher-Cryptomator
+    '';
+  };
+
+  # perform fake build to make a fixed-output derivation out of the files downloaded from maven central (120MB)
+  deps = stdenv.mkDerivation {
+    name = "cryptomator-${version}-deps";
+    inherit src;
+
+    buildInputs = [ openjdk maven ];
+
+    buildPhase = ''
+      cd main
+      while mvn -Prelease package -Dmaven.repo.local=$out/.m2 -Dmaven.wagon.rto=5000; [ $? = 1 ]; do
+        echo "timeout, restart maven to continue downloading"
+      done
+    '';
+
+    # keep only *.{pom,jar,sha1,nbm} and delete all ephemeral files with lastModified timestamps inside
+    installPhase = ''
+      find $out/.m2 -type f -regex '.+\(\.lastUpdated\|resolver-status\.properties\|_remote\.repositories\)' -delete
+      find $out/.m2 -type f -iname '*.pom' -exec sed -i -e 's/\r\+$//' {} \;
+    '';
+
+    outputHashAlgo = "sha256";
+    outputHashMode = "recursive";
+    outputHash = "15gwldn9dc4gxi8ja4lay5qdadvhxry17ggj3cqamk8g38rh97v0";
+  };
+
+in stdenv.mkDerivation rec {
+  inherit pname;
+  inherit version;
+  inherit src;
+
+  buildPhase = ''
+    cd main
+    mvn -Prelease package --offline -Dmaven.repo.local=$(cp -dpR ${deps}/.m2 ./ && chmod +w -R .m2 && pwd)/.m2
+  '';
+
+  installPhase = ''
+    mkdir -p $out/usr/share/cryptomator/libs/
+    mkdir -p $out/bin/
+
+    cp buildkit/target/libs/* buildkit/target/linux-libs/* $out/usr/share/cryptomator/libs/
+
+    cat > $out/usr/share/cryptomator/cryptomator <<EOF
+    #!/usr/bin/env sh
+    exec ${openjdk}/bin/java -verbose \
+        -cp "$out/usr/share/cryptomator/libs/*" \
+        -Dcryptomator.settingsPath="~/.config/Cryptomator/settings.json" \
+        -Dcryptomator.ipcPortPath="~/.config/Cryptomator/ipcPort.bin" \
+        -Dcryptomator.logDir="~/.local/share/Cryptomator/logs" \
+        -Dcryptomator.mountPointsDir="~/.local/share/Cryptomator/mnt" \
+        -Djdk.gtk.version=2 \
+        -Xss20m \
+        -Xmx512m \
+        org.cryptomator.launcher.Cryptomator
+EOF
+
+    chmod +x $out/usr/share/cryptomator/cryptomator
+    ln -s $out/usr/share/cryptomator/cryptomator $out/bin/cryptomator
+
+    wrapProgram "$out/usr/share/cryptomator/cryptomator" \
+     --prefix PATH : "$out/usr/share/cryptomator/libs/:${lib.makeBinPath [ openjdk glib ]}" \
+     --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath [ fuse ]}" \
+     --set JAVA_HOME "$openjdk"
+
+    # install desktop entry
+    install -Dm644 -t $out/share/applications ${desktopItem}/share/applications/*
+    install -Dm644 "${./org.cryptomator.Cryptomator.svg}" "$out/share/icons/hicolor/scalable/apps/org.cryptomator.Cryptomator.svg"
+    install -Dm644 "${./org.cryptomator.Cryptomator.png}" "$out/share/icons/hicolor/512x512/apps/org.cryptomator.Cryptomator.png"
```

Comment:
I agree that those should be fetched from a URL. They could e.g. be extracted from the AppImage like the AUR package does (https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=cryptomator) but we should even be able to get them without having to fetch the AppImage (and since the AppImage asset is hopefully built via CI/CD that should reveal where the files come from).

> In the best case they would be part of the source itself, so maybe this should be fixed in their source repo?

Seems like they already have quite a lot of PNG files in their repo (https://github.com/cryptomator/cryptomator/search?q=extension%3Apng and there's also https://github.com/cryptomator/cryptomator.github.io/tree/develop/static/img) so if the required two files aren't already there it might be a good idea to open an upstream issue (this part should at least be documented for packagers).

Reply:
@primeos I don't think the AUR package gets it from the AppImage as they are checked in there too.

There is a png available in the source that would fit. I will just drop the svg and only provide the png.

Reply:
Files are now taken from https://github.com/cryptomator/cryptomator-linux/tree/master/resources/appimage/AppDir/